### ISSUE TITLE
Improper cleanup of files created with File::Edit::Portable::tempfile()

### ISSUE THREAD

**Issue by XSven** [stevieb9/devel-examine-subs#31]
`Devel::Examine::Subs` uses `File::Edit::Portable::tempfile()` in line

`+755 my $tempfile = $self->{rw}->tempfile;`

 to create a temporary file. As far as I have understood the implementation of `File::Edit::Portable::tempfile()` no auto-cleanup is implemented. Therefore the cleanup has to be manually implemented somewhere after line

`+767  close $tempfile;`

**Comment by jsf116** [comment]
As a matter of fact this is also for me a severe issue because in case of really frequent application of Devel::Examine::Subs such leavings are nothing other than a kind of leak and hence is a potential damage for the whole system.
A timely bug fix would be greatly appreciated!

**Comment by stevieb9** [comment]
I've implemented a fix for this. Is there any chance either one of you could clone the repo (https://github.com/stevieb9/devel-examine-subs) and test it locally before I upload a distribution to the CPAN?

I'd like to ensure that it does in fact correct your problem and doesn't cause any negative side effects before public release.

**Comment by XSven** [comment]
I have cloned the commit afc2231 on master branch and processed all unit tests. All tests were ok but they have created 260 tempfiles with the contend "abc\n" in TMPDIR that were not properly cleaned.

**Comment by stevieb9** [comment]
I see that's caused by a bug in `File::Edit::Portable`.

See https://github.com/stevieb9/file-edit-portable/issues/37

I'll get a fix out for that and ensure that everything is cleaned up after itself.

**Comment by stevieb9** [comment]
I have figured this out. It was definitely in `File::Edit::Portable`, and I've corrected the issue. If you install the master branch of [File::Edit::Portable](https://github.com/stevieb9/file-edit-portable) (`https://github.com/stevieb9/file-edit-portable`), your problems should go away:

```
spek@scelia ~/repos/devel-examine-subs $ ll /tmp | wc -l
315

spek@scelia ~/repos/devel-examine-subs $ make test

...

All tests successful.
Files=66, Tests=1580, 16 wallclock secs ( 0.27 usr  0.07 sys + 15.00 cusr  1.40 csys = 16.74 CPU)
Result: PASS

spek@scelia ~/repos/devel-examine-subs $ ll /tmp | wc -l
315
```

Please test again, and if all seems well, I'll publish the new versions of both distributions to the CPAN.

**Comment by XSven** [comment]
I have tested File::Edit::Portable 1.25 and 1.26 (not yet released) successfully. I have tested Devel::Examine::Subs 1.72 (not yet released) against File::Edit::Portable 1.26 successfully. Auto-cleanup works as expected. Please publish the not yet released versions to CPAN.

**Comment by stevieb9** [comment]
Awesome. Thank you very much for your patience and your testing.

The two new distributions are now indexed on the CPAN.
